src/include/detail/linalg/tdb_io.h

```diff
@@ -376,63 +415,40 @@ auto sizes_to_indices(const std::vector<T>& sizes) {
  */
 template <class T>
 auto read_bin_local(const std::string& bin_file, size_t subset = 0) {
-  if (!std::filesystem::exists(bin_file)) {
-    throw std::runtime_error("file " + bin_file + " does not exist");
-  }
```

Comment:
(Minor) What's the reasoning to remove?

Reply:
The std::fstream{} constructor already checks if that file exists, so I thought that this code is redundant, meaning it would be checked twice. (No one guarantees that between filestystem::exists and opening the file, the file would not be deleted). File paths are not atomic.